pkg/controllers/management/aks/aks_cluster_handler.go

```diff
@@ -148,11 +145,18 @@ func (e *aksOperatorController) onClusterChange(key string, cluster *mgmtv3.Clus
 		logrus.Infof("waiting for cluster AKS [%s] create failure to be resolved", cluster.Name)
 		return e.SetFalse(cluster, apimgmtv3.ClusterConditionProvisioned, failureMessage)
 	case "active":
+		// If an imported cluster, set the upstream spec and update the AKSClusterConfig object with fields from the
+		// cluster state. This will prevent configuration in Azure from getting overwritten in later updates. Non
+		// imported clusters will already have upstream spec set.
 		if cluster.Spec.AKSConfig.Imported {
 			if cluster.Status.AKSStatus.UpstreamSpec == nil {
-				// non imported clusters will have already had upstream spec set
 				return e.setInitialUpstreamSpec(cluster)
 			}
+			if !reflect.DeepEqual(cluster.Status.AKSStatus.UpstreamSpec, cluster.Spec.AKSConfig) {
```

Comment:
Is it okay to use DeepEqual here? And do we want to copy upstream spec to AKSConfig? Both of these bypass the idea of leaving fields nil that are unmanaged by rancher. This is vital to syncing the state from upstream. How does this prevent us from overwriting upstream fields? I would think it would do the opposite.

Reply:
The UI should be showing the fields as follows: upstreamspec, then if any fields on spec are non nil, overwrite the field on upstreamspec with that value.